### DISCUSSION TITLE
Send IFormFileCollection

### DISCUSSION THREAD

**Discussion by Giorgobiani-G** [canton7/RestEase#267] [General]
```cs
[HttpPost]
public async Task UploadAsync([FromForm] int sourceId,
                              [FromForm] int documentTypeId,
                              [FromForm] string documentTypeName,
                              [FromForm] long associationId,
                              IFormFileCollection files,
                              CancellationToken cancellationToken = default)
{
    

    var data = new Dictionary<string, object>
    {
     { "SourceId", sourceId},
     { "DocumentTypeId", documentTypeId},
     { "DocumentTypeName", documentTypeName },
     { "AssociationId", associationId },
     { "Files", files  }
    };

    await _fileStorageClient.Upload(data, cancellationToken);
}
```

i have controller getting data [FromForm], how can i send this form to another api that expects the same data the same fomat.

I have defined api client like this and here i pass data from contreoller:

```cs
public interface IFileStorageClient
{
    [Post("file/upload")]
    Task Upload([Body(BodySerializationMethod.UrlEncoded)] Dictionary<string, object> data, CancellationToken cancellationToken);
}
```

but when i send request api returns 415 (Unsupported Media Type). can u help me?

**Comment by canton7** [comment]
There aren't many details here. Do you need to set a `Content-Type` header (that's normally what 415 means)?

**Reply by Giorgobiani-G** [reply]
the second apis endpoint expecting the fields exactly the same as here: [FromForm] int sourceId,
                              [FromForm] int documentTypeId,
                              [FromForm] string documentTypeName,
                              [FromForm] long associationId,
                              IFormFileCollection files,
![Screenshot (270)](https://github.com/canton7/RestEase/assets/63700331/4b1fca62-0a87-457f-9828-78f780eef317)
here is it .

how can i send the same parameters from restease api?

**Reply by Giorgobiani-G** [reply]
also when selecting image from file and sending request ,  checking the Content type header of IFormFile it has value "image/png" .
